include/zephyr/net/socket.h

```diff
@@ -171,6 +165,13 @@ struct zsock_pollfd {
  */
 #define TLS_SESSION_CACHE_PURGE 13
 
+/** Socket option to set DTLS Connection ID to be used for the DTLS session.
+ *  The option accepts an byte array, holding the CID to use.
+ *  Setting an empty CID (option length set to 0) indicates that the socket is
+ *  willing to handle CID from a peer, but does not specify its own CID.
+ */
+#define TLS_DTLS_CONNECTION_ID 14
```

Comment:
This change should be squashed into https://github.com/zephyrproject-rtos/zephyr/pull/44914/commits/0ea39ec26023a2cdbdbc5674cb0e35d6461ed502.